src/main/java/com/google/api/generator/engine/ast/LambdaExpr.java

```diff
@@ -26,11 +26,7 @@
 @AutoValue
 public abstract class LambdaExpr implements Expr {
   @Override
-  public TypeNode type() {
-    // TODO(v2): Support set of FunctionalInterface  parameterized on the args and return type,
-    // which would enable assignment to an appropriate variable.
-    return TypeNode.VOID;
-  }
+  public abstract TypeNode type();
```

Comment:
keeping this despite sonar complains about already defined in `Expr` interface. It's seems the recommend way for @AutoValue as explained in [this doc](https://github.com/google/auto/blob/master/value/userguide/howto.md#-have-autovalue-implement-a-concrete-or-default-method). Also keeps consistent with other expression classes like [`ValueExpr`](https://github.com/googleapis/gapic-generator-java/blob/main/src/main/java/com/google/api/generator/engine/ast/ValueExpr.java#L21), [`CastExpr`](https://github.com/googleapis/gapic-generator-java/blob/main/src/main/java/com/google/api/generator/engine/ast/CastExpr.java#L23)

Reply:
If you want, we can still expose the `setType()` method and set a default value in the `builder()` method, but like you said `ReturnExpr().expr()` should always have `type` associated so we should never call `setType()` explicitly.